# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/MozContainerWayland.cpp
# Commit: 2b20b560fe61
# Full Hash: 2b20b560fe614c26fe779367c3323addaa55122a
# Author: stransky <stransky@redhat.com>
# Date: 2021-06-20 09:32:48
# Description:
#   Bug 1716796 [Wayland] Poll mozcontainer remap state, r=rmader
#   
#   - Poll mozcontainer remap state by moz_container_wayland_remapped()
#   - Remove WindowSurface::Reset()
#   - Remove moz_container_wayland_set_window_surface() as it adds extra ref at WindowSurface() and causes cyclic dependency.
# ==============================================================================

diff -r 668013f3e201 -r 2b20b560fe61 widget/gtk/MozContainerWayland.cpp
--- a/widget/gtk/MozContainerWayland.cpp	Sat Jun 19 07:06:41 2021 +0300
+++ b/widget/gtk/MozContainerWayland.cpp	Sat Jun 19 09:33:19 2021 +0000
@@ -180,6 +180,7 @@
   container->opaque_region_subtract_corners = false;
   container->opaque_region_used = false;
   container->surface_needs_clear = true;
+  container->container_remapped = true;
   container->subsurface_dx = 0;
   container->subsurface_dy = 0;
   container->before_first_size_alloc = true;
@@ -297,6 +298,7 @@
   wl_container->surface_needs_clear = true;
   wl_container->ready_to_draw = false;
   wl_container->buffer_scale = 1;
+  wl_container->container_remapped = true;
 }
 
 static gboolean moz_container_wayland_map_event(GtkWidget* widget,
@@ -505,10 +507,6 @@
   }
   wl_subsurface_set_desync(wl_container->subsurface);
 
-  if (wl_container->window_surface) {
-    wl_container->window_surface->Reset();
-  }
-
   // Try to guess subsurface offset to avoid potential flickering.
   int dx, dy;
   if (moz_container_get_nsWindow(container)->GetCSDDecorationOffset(&dx, &dy)) {
@@ -601,6 +599,12 @@
   return ret;
 }
 
+gboolean moz_container_wayland_get_and_reset_remapped(MozContainer* container) {
+  int ret = container->wl_container.container_remapped;
+  container->wl_container.container_remapped = false;
+  return ret;
+}
+
 void moz_container_wayland_update_opaque_region(MozContainer* container,
                                                 bool aSubtractCorners) {
   MozContainerWayland* wl_container = &container->wl_container;
@@ -638,9 +642,3 @@
   }
   return wl_container->viewport;
 }
-
-void moz_container_wayland_set_window_surface(
-    MozContainer* container, RefPtr<WindowSurface> window_surface) {
-  MozContainerWayland* wl_container = &container->wl_container;
-  wl_container->window_surface = window_surface;
-}